File: java/client/src/test/java/org/signal/libsignal/net/UnauthCallQualityServiceTest.kt
//
// Copyright 2026 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.libsignal.net

import kotlinx.coroutines.test.runTest
import org.signal.libsignal.internal.NativeTestingNice
import kotlin.test.Test
import kotlin.test.assertIs

class UnauthCallQualityServiceTest {
  @Test
  fun testSubmitCallQualitySurvey() =
    runTest {
      GrpcTestCase.runTests(
        NativeTestingNice.TESTING_SubmitCallQualitySurveyTests(),
        { tokio, listener ->
          UnauthenticatedChatConnection.fakeConnect(tokio, listener, Network.Environment.STAGING)
        },
        ::UnauthCallQualityService,
        invoke = { chat, req ->
          chat.submitCallQualitySurvey(
            survey = req,
          )
        },
        check = { _, actual ->
          assertIs<RequestResult.Success<Unit>>(actual)
        },
      )
    }
}

File: node/ts/test/chat/UnauthCallQualityServiceTest.ts
//
// Copyright 2026 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import { config, use } from 'chai';
import chaiAsPromised from 'chai-as-promised';

import * as NativeNice from '../../NativeNice.js';
import * as util from '../util.js';
import { UnauthCallQualityService, CallQualitySurvey } from '../../net.js';
import { connectUnauth, defineTestGrpcCases } from './ServiceTestUtils.js';

use(chaiAsPromised);

util.initLogger();
config.truncateThreshold = 0;

describe('UnauthCallQualityService', () => {
  describe('submitCallQualitySurvey', () => {
    defineTestGrpcCases(
      NativeNice.TESTING_SubmitCallQualitySurveyTests(),
      connectUnauth<UnauthCallQualityService>,
      async (
        chat: UnauthCallQualityService,
        survey: CallQualitySurvey,
        _resp: void
      ) => {
        await chat.submitCallQualitySurvey({ survey });
      }
    );
  });
});
